WCFDataService/Client/System/Data/Services/Client/Materialization/EntryValueMaterializationPolicy.cs

```diff
@@ -510,7 +510,11 @@ private object GetOrCreateCollectionProperty(object instance, ClientPropertyAnno
                         collectionType = typeof(System.Collections.ObjectModel.Collection<>).MakeGenericType(property.EntityCollectionItemType);
                     }
 
-                    result = this.CreateNewInstance(property.EdmProperty.Type, collectionType);
+                    // Support proxies with uninitialized DataServiceCollection<> type properties.
```

Comment:
The code path (more specifically: result = Util.ActivatorCreateInstance(collectionType, new object[] { null, TrackingMode.None });) will be hit when the proxy does NOT use lazy DSC<> property construction but has them uninitialized (null).

The following query triggers this:
`            var q = from f in ctx.Files.Expand("Parties").AddQueryOption("$select", "Parties")
                    where f.Parties.Any()
                    select f;
`
Parties is of type DSC<> and it is initialized with null and does not auto/layzy construct in the property getter.

I included this change for completeness, as the default DSC<> constructor is very different from the overload with non tracking.